crates/ruff_python_parser/tests/generate_inline_tests.rs

```diff
@@ -0,0 +1,259 @@
+//! This module takes specially formatted comments from `ruff_python_parser` code
+//! and turns them into test fixtures. The code is derived from `rust-analyzer`
+//! and `biome`.
+//!
+//! References:
+//! - <https://github.com/rust-lang/rust-analyzer/blob/e4a405f877efd820bef9c0e77a02494e47c17512/crates/parser/src/tests/sourcegen_inline_tests.rs>
+//! - <https://github.com/biomejs/biome/blob/b9f8ffea9967b098ec4c8bf74fa96826a879f043/xtask/codegen/src/parser_tests.rs>
+use std::collections::HashMap;
+use std::fs;
+use std::ops::{Deref, DerefMut};
+use std::path::{Path, PathBuf};
+
+use anyhow::{Context, Result};
+use itertools::Itertools;
+
+fn project_root() -> PathBuf {
+    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("../../")
+        .canonicalize()
+        .unwrap()
+}
+
+#[test]
+fn generate_inline_tests() -> Result<()> {
+    let parser_dir = project_root().join("crates/ruff_python_parser/src/parser");
+    let tests = TestCollection::try_from(parser_dir.as_path())?;
+
+    install_tests(&tests.ok, "crates/ruff_python_parser/resources/inline/ok")?;
+    install_tests(&tests.err, "crates/ruff_python_parser/resources/inline/err")?;
+
+    Ok(())
+}
+
+fn install_tests(tests: &HashMap<String, Test>, target_dir: &str) -> Result<()> {
+    let root_dir = project_root();
+    let tests_dir = root_dir.join(target_dir);
+    if !tests_dir.is_dir() {
+        fs::create_dir_all(&tests_dir)?;
+    }
+
+    // Test kind is irrelevant for existing test cases.
+    let existing = existing_tests(&tests_dir)?;
+
+    let unreferenced = existing
+        .iter()
+        .filter(|&(name, _)| !tests.contains_key(name))
+        .map(|(_, path)| path)
+        .collect::<Vec<_>>();
+
+    if !unreferenced.is_empty() {
+        anyhow::bail!(
+            "Unreferenced test files found for which no comment exists:\n  {}\n",
+            unreferenced
+                .iter()
+                .map(|path| path.strip_prefix(&root_dir).unwrap().display())
+                .join("\n  ")
+        );
```

Comment:
Can we extend the error message with what needs to be done in this case? Is there a specific command that i need to run? Should I delete the test?

Reply:
Updated the message with "Please delete these files manually"